gems/activerecord/6.0/activerecord.rbs

```diff
@@ -722,10 +722,702 @@ end
 
 module ActiveRecord
   module Associations
-    class CollectionProxy
-      def build: (?::Hash[untyped, untyped] attributes) -> untyped | ...
-      def create: (?::Hash[untyped, untyped] attributes) -> untyped | ...
-      def create!: (?::Hash[untyped, untyped] attributes) -> untyped | ...
+    # Collection proxies in Active Record are middlemen between an
+    # <tt>association</tt>, and its <tt>target</tt> result set.
+    #
+    # For example, given
+    #
+    #   class Blog < ActiveRecord::Base
+    #     has_many :posts
+    #   end
+    #
+    #   blog = Blog.first
+    #
+    # The collection proxy returned by <tt>blog.posts</tt> is built from a
+    # <tt>:has_many</tt> <tt>association</tt>, and delegates to a collection
+    # of posts as the <tt>target</tt>.
+    #
+    # This class delegates unknown methods to the <tt>association</tt>'s
+    # relation class via a delegate cache.
+    #
+    # The <tt>target</tt> result set is not loaded until needed. For example,
+    #
+    #   blog.posts.count
+    #
+    # is computed directly through SQL and does not trigger by itself the
+    # instantiation of the actual post records.
+    class CollectionProxy < Relation
+      interface _EachPair
+        def each_pair: () { ([untyped, untyped]) -> untyped } -> self
+        def each: () { ([untyped, untyped]) -> untyped } -> self
+        def empty?: () -> bool
+      end
+
+      def initialize: (untyped klass, untyped association) -> untyped
```

Comment:
@ksss Is it better not to move these definitions unrelated to this change?

Reply:
Yes. review burden will be reduced.

Reply:
Done!